### PR TITLE
Fix Contents paths macOS

### LINKED ISSUE #4761: macOS: wrong path for objects PROTO controller paths
**Describe the Bug**
Paths for controllers in the `objects` directory are not found correctly in the macOS .dmg package.
The `Contents` directory is missing in the path.
The path for compiling controllers is also wrong and does not contain `Contents`.

### PR DESCRIPTION
**Pull request by ThomasOliverKimble**
**Description**
Fixes #4761 and updates documentation.

### CODE REVIEW

File: src/webots/vrml/WbProtoModel.cpp [hunk 1]
```diff
@@ -454,6 +454,10 @@ const QString WbProtoModel::projectPath() const {
     if (mPath.startsWith("https://"))
       path = path.replace(QRegularExpression("https://raw.githubusercontent.com/cyberbotics/webots/[a-zA-Z0-9\\_\\-\\+]+/"),
                           WbStandardPaths::webotsHomePath());
+#ifdef __APPLE__
+    if (path.indexOf(WbStandardPaths::webotsHomePath()) >= 0)
```
[the remark this answers is not in the document]
Comment (ThomasOliverKimble) [reply src/webots/vrml/WbProtoModel.cpp:458]: Yes it should never happen. I will change to != 0.

File: resources/Makefile.java.include [hunk 2]
```diff
@@ -38,7 +38,7 @@ endif
 ifndef WEBOTS_HOME_PATH
  null :=
  space := $(null) $(null)
- WEBOTS_HOME_PATH=$(subst $(space),\ ,$(strip $(subst \,/,$(WEBOTS_HOME))))
+ WEBOTS_HOME_PATH?=$(subst $(space),\ ,$(strip $(subst \,/,$(WEBOTS_HOME))))
```
Comment (omichel) [inline resources/Makefile.java.include:41]: This seems unneeded as there is a `ifndef WEBOTS_HOME_PATH` statement above.

File: scripts/install/bash_profile.mac [hunk 3]
```diff
@@ -17,4 +17,5 @@ export WEBOTS_ALLOW_MODIFY_INSTALLATION=1                 # If defined, you are
 # These environment variables are necessaries to compile and run Webots #
 #########################################################################
 
-export WEBOTS_HOME=$HOME/webots          # Defines the path to Webots home.
+export WEBOTS_HOME=$HOME/webots                        # Defines the path to Webots home.
+export WEBOTS_HOME_PATH=$WEBOTS_HOME/Contents          # Defines the path to Webots home.
```
Comment (omichel) [inline scripts/install/bash_profile.mac:21]:
```suggestion
export WEBOTS_HOME_PATH=$WEBOTS_HOME/Contents          # Defines the path to Webots home Contents folder (used for finding include files when compiling from a terminal)
```

File: src/webots/vrml/WbProtoModel.cpp [hunk 4]
```diff
@@ -454,6 +454,10 @@ const QString WbProtoModel::projectPath() const {
     if (mPath.startsWith("https://"))
       path = path.replace(QRegularExpression("https://raw.githubusercontent.com/cyberbotics/webots/[a-zA-Z0-9\\_\\-\\+]+/"),
                           WbStandardPaths::webotsHomePath());
+#ifdef __APPLE__
+    if (path.indexOf(WbStandardPaths::webotsHomePath()) == 0)
+      path.insert(path.indexOf(WbStandardPaths::webotsHomePath()) + WbStandardPaths::webotsHomePath().length(), "Contents/");
```
Comment (omichel) [inline src/webots/vrml/WbProtoModel.cpp:459]: Simpler this way:
```suggestion
    if (path.startsWith(WbStandardPaths::webotsHomePath()))
      path.insert(WbStandardPaths::webotsHomePath().length(), "Contents/");
```

**Review by omichel — CHANGES_REQUESTED** [review]

File: docs/guide/compiling-controllers-in-a-terminal.md [hunk 5]
```diff
@@ -19,11 +19,12 @@ On macOS, type this:
 
 ```sh
 $ export WEBOTS_HOME=/Applications/Webots.app
+$ export WEBOTS_HOME_PATH=$WEBOTS_HOME/Contents
 ```
 
 Or add this line to your "~/.profile" file.
```
Comment (omichel) [inline docs/guide/compiling-controllers-in-a-terminal.md:25]:
```suggestion
Or add these lines to your "~/.profile" file.
```

File: docs/guide/compiling-controllers-in-a-terminal.md [hunk 6]
```diff
@@ -19,11 +19,12 @@ On macOS, type this:
...
 
-Once `WEBOTS_HOME` is defined, you should be able to compile in a terminal, with the `make` command.
+Once `WEBOTS_HOME` and `WEBOTS_HOME_PATH` are defined, you should be able to compile in a terminal, with the `make` command.
```
Comment (omichel) [inline docs/guide/compiling-controllers-in-a-terminal.md:27]:
```suggestion
Once these environment variables are defined, you should be able to compile in a terminal, with the `make` command.
```

File: src/webots/vrml/WbProtoModel.cpp [hunk 7]
```diff
@@ -454,6 +454,10 @@ const QString WbProtoModel::projectPath() const {
     if (mPath.startsWith("https://"))
       path = path.replace(QRegularExpression("https://raw.githubusercontent.com/cyberbotics/webots/[a-zA-Z0-9\\_\\-\\+]+/"),
                           WbStandardPaths::webotsHomePath());
+#ifdef __APPLE__
+    if (path.startsWith(WbStandardPaths::webotsHomePath()))
+      path.insert(path.indexOf(WbStandardPaths::webotsHomePath()) + WbStandardPaths::webotsHomePath().length(), "Contents/");
```
Comment (omichel) [inline src/webots/vrml/WbProtoModel.cpp:459]:
```suggestion
      path.insert(WbStandardPaths::webotsHomePath().length(), "Contents/");
```

**Review by omichel — CHANGES_REQUESTED** [review]

**Review by omichel — COMMENTED** [review]
Thank you.

**Review by omichel — APPROVED** [review]

**Pull request merged**